src/server/mod.ts

```diff
@@ -131,11 +152,34 @@ export async function start(routes: Manifest, opts: StartOptions = {}) {
   }
 }
 
-async function bootServer(handler: ServeHandler, opts: StartOptions) {
-  if (opts.experimentalDenoServe === true) {
+function modeTls(opts: StartOptions | StartTlsOptions): boolean {
+  return !!(opts as StartTlsOptions)?.keyFile ||
+    !!(opts as StartTlsOptions)?.certFile ||
+    !!(opts as StartTlsOptions)?.cert || !!(opts as StartTlsOptions)?.key;
+}
+
+async function bootServer(
+  handler: ServeHandler,
+  opts: StartOptions | StartTlsOptions,
+) {
+  if (opts.experimentalDenoServe) {
     // @ts-ignore as `Deno.serve` is still unstable.
     await Deno.serve({ ...opts, handler }).finished;
   } else {
     await serve(handler, opts);
   }
 }
+
+async function bootServerTls(handler: ServeHandler, opts: StartTlsOptions) {
+  if (opts.experimentalDenoServe) {
+    if (!!opts.certFile || !!opts.keyFile) {
```

Comment:
Is the logic here correct? Currently it's if either `opts.certFile` or `opts.keyFile` is truthy, throw an `UnstableFeatureError`. I'm assuming this is meant to check if either of those options haven't been set, throw an error.